### ISSUE TITLE
File list truncated

### ISSUE THREAD

**Issue by CallMeLaNN** [cline/cline#541]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude 3.5 Sonnet

### What happened?

For a large, monorepo or project like tauri (rust + web codes), it is unnecessary to send list of all files under `Current Working Directory` of initial API request.

I think it already list only source, could be obey the `.gitignore` too 🤷‍♂️. However the list is quite long for real project that includes irrelevant contexts to the prompt.

Consider adding cline ignore or simply don't send `Current Working Directory` by default and let me add the necessary folder into the prompt.

Other than to **save** tokens for a long list of files, this fix can improve result by including **only relevant list** of files that might be **truncated** at the bottom.

### Steps to reproduce

1. Open large project
2. Have some prompt and run
3. Open the "v" icon besides "API Request" to view full request prompt
4. See the `# Current Working Directory ...` section

### Relevant API REQUEST output

```shell
<task>
Prompt...
</task>

<file_content path="src/store/currentFile.test.ts">
...
</file_content>

<environment_details>
# VSCode Visible Files
../../../../../response_11/1
../../../../../response_11/1
../../../../../response_11/0
../../../../../response_11/2
../../../../../response_11/2
../../../../../response_11/0
src/store/currentFile.test.ts

# VSCode Open Tabs
Todo.md # Unnecessary open tabs
../../../../../Untitled-1 Unnecessary open tabs
src/...

# Current Working Directory (/Users/user/Documents/code/project-abc) Files
.gitignore
package.json
...
playwright/
playwright-report/
public/...
src/...
src-tauri/...
src-tauri/target/
storybook-static/
test-results/

(File list truncated. Use list_files on specific subdirectories if you need to explore further.)
</environment_details>
```

### Additional context

The list is actually long, I truncate with ...
- public/... Bunch of HTML and CSS in public folder
- src-tauri/... Rust files that is not related to the prompt
- some static outputs
- some test results

**Comment by Purecode24** [comment]
I agree, at least we need an option to switch if off a it consumes a huge chunk of input precious input context.

**Comment by Leibnizhu** [comment]
agree, it's unnecessary and may cause some llm hallucination

**Comment by DaleLJefferson** [comment]
This is such an important feature, just let the model query the file tree if it needs to see it. Saying Hi does not need the whole (or truncated) fill tree.

**Comment by DaleLJefferson** [comment]
Roo does this really nicely, allow the user to choose, I just wish they had an "unlimited" option instead of maxing out at 500

With these 1M context size models more context is often better...

<img width="577" alt="Image" src="https://github.com/user-attachments/assets/f9431ee7-3768-43bf-86de-8de07495e534" />

**Comment by rarensu** [comment]
.clineignore doesn't even hide the files in subdirectories, it just adds the lock emoji. 
```
# Current Working Directory 
...
🔒 references/
🔒 references/bubblewrap/
🔒 references/bubblewrap/.dir-locals.el
...
(File list truncated. Use list_files on specific subdirectories if you need to explore further.)
```

Can we at least make it so that the current working directory list does NOT expand into ignored directories? 

As a silly workaround, I am thinking I might remove read permission on the directories to stop the Cline extension from trying to search them.

**Comment by lishenxydlgzs** [comment]
Any updates on actually removing the .clineignore'd files from the prompt? I am working with directories with a lot of generated content which are not necessary in understanding the code base. Showing the ignored file/dirs just eats up tokens without actual benefit and it slows Cline down since otherwise Cline knows the useful directory structure upfront without having to use list_files.

**Comment by celestial-vault** [comment]
Thanks for the suggestion! This has been consolidated into #4364 which addresses the broader problem of configurable file listing behavior in environment details.
